File: Week 5/printer/hyewon.cpp
#include <string>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int solution(vector<int> priorities, int location) {
	int answer = 0;
	queue<pair<int, int>> q;

	for (int i = 0; i < priorities.size(); i++)
		q.push({ i, priorities[i] });

	sort(priorities.begin(), priorities.end());

	int cnt = 0;
	while (!q.empty()) {
		if (q.front().second != priorities.back()) {
			q.push(q.front());
			q.pop();
		}
		else {
			cnt++;
			if (q.front().first == location) {
				answer = cnt; break;
			}
			else {
				priorities.pop_back(); q.pop();
			}
		}
	}

	return answer;
}

File: Week10/LifeBoat/hyewon.py
from collections import deque
def solution(people, limit):
    answer = 0
    people.sort()
    people = deque(people)
    while len(people) != 0:
        if len(people) == 1:
            answer += 1
            break
        else:
            if people[0] + people.pop() <= limit:
                people.popleft()
        answer += 1

    return answer

print(solution([70, 50, 80, 50], 100))
print(solution([70, 80, 50], 100))